src/devices/Common/Iot/Device/Multiplexing/GpioOutputSegment.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.Gpio;
+using System.Diagnostics;
+using System.Threading;
+using Iot.Device.Multiplexing.Utility;
+
+namespace Iot.Device.Multiplexing
+{
+    /// <summary>
+    /// IOutputSegment implementation that uses GpioController.
+    /// </summary>
+    public class GpioOutputSegment : IOutputSegment, IDisposable
+    {
+        private readonly int[] _pins;
+        private readonly bool _shouldDispose;
+        private readonly VirtualOutputSegment _segment;
```

Comment:
What purpose does the `VirtualOutputSegment` have here? No value is ever read from it.

Reply:
That's not true. It is used extensively.

Reply:
`VirtualOutputSegment` maintains state for all the writes.